src/messages/mod.rs

```diff
@@ -231,14 +275,77 @@ impl Debug for SecurityMetadata {
 }
 
 /// Wrapper around a routing message, signed by the originator of the message.
-#[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+pub struct PeekedSignedRoutingMessageInfo {
+    /// Destination location
+    pub dst: Location,
+}
+
+impl<'de> Deserialize<'de> for PeekedSignedRoutingMessageInfo {
+    fn deserialize<D: Deserializer<'de>>(deserialiser: D) -> ::std::result::Result<Self, D::Error> {
+        struct PeekVisitor;
+
+        impl<'de> de::Visitor<'de> for PeekVisitor {
+            type Value = PeekedSignedRoutingMessageInfo;
+
+            fn expecting(&self, formatter: &mut Formatter) -> fmt::Result {
+                formatter.write_str("expecting to find SignedRoutingMessage and getting field of PeakedSignedRoutingMessageInfo")
+            }
+
+            fn visit_seq<V>(
+                self,
+                mut seq: V,
+            ) -> ::std::result::Result<PeekedSignedRoutingMessageInfo, V::Error>
+            where
+                V: de::SeqAccess<'de>,
+            {
+                let dst = seq
+                    .next_element()?
+                    .ok_or_else(|| de::Error::invalid_length(0, &self))?;
+                Ok(PeekedSignedRoutingMessageInfo { dst })
+            }
+        }
+        deserialiser.deserialize_tuple(1, PeekVisitor)
+    }
+}
+
+/// Wrapper around a routing message, signed by the originator of the message.
+/// Serialized as simple tupple to ease partial deserialization.
+#[derive(Eq, PartialEq, Clone, Hash)]
 pub struct SignedRoutingMessage {
     /// A request or response type message.
     content: RoutingMessage,
     /// Optional metadata for verifying the sender
     security_metadata: SecurityMetadata,
 }
 
+impl Serialize for SignedRoutingMessage {
+    fn serialize<S: Serializer>(&self, serialiser: S) -> ::std::result::Result<S::Ok, S::Error> {
+        (
+            &self.content.dst,
+            &self.content.src,
+            &self.content.content,
+            &self.security_metadata,
+        )
+            .serialize(serialiser)
+    }
+}
+
+impl<'de> Deserialize<'de> for SignedRoutingMessage {
+    fn deserialize<D: Deserializer<'de>>(deserialiser: D) -> ::std::result::Result<Self, D::Error> {
+        let (dst, src, content, security_metadata): (
+            Location,
+            Location,
+            MessageContent,
+            SecurityMetadata,
+        ) = Deserialize::deserialize(deserialiser)?;
+        Ok(Self {
+            content: RoutingMessage { dst, src, content },
```

Comment:
RoutingMessage is defined as ` {src, dst, content}` at https://github.com/maidsafe/routing/blob/fleming/src/messages/mod.rs#L477-L484 ?

Reply:
it does not matter, since these are named fields, but I can fix.